Testing/MeshTests/MeshTests.cpp

```diff
@@ -295,3 +299,50 @@ TEST(MeshTests, icpTest)
 
   ASSERT_TRUE(source == ground_truth);
 }
+
+
+TEST(MeshTests, warpTest1)
```

Comment:
These tests are still pretty big. One thing that could be done to improve them is to add more assertions throughout. Otherwise if the test fails there could be a dozen different places where something went wrong.